onnxruntime/contrib_ops/cpu/skip_layer_norm.cc

```diff
@@ -118,12 +173,15 @@
         }
 
         for (int64_t h = 0; h < hidden_size; h++) {
+          DoubleOrFloat output_value = ConvertMLFloat16ToDoubleOrFloatIfNeeded<T, DoubleOrFloat>(p_output[h]);
+          DoubleOrFloat gamma_value = ConvertMLFloat16ToDoubleOrFloatIfNeeded<T, DoubleOrFloat>(gamma_data[h]);
           if (simplified) {
-            p_output[h] = p_output[h] / mean_square * gamma_data[h];
+            p_output[h] = ConvertDoubleOrFloatToMLFloat16IfNeeded<T>(output_value / mean_square * gamma_value);
```

Comment:
the p_output was converted twice. fp16/fp32 conversion is heavy. maybe it is faster to convert the whole input, p_skip to a tempory buffer, do computation with fp32 and convert the result back to p_output

Reply:
updated in the latest iteration